vars/deploy.groovy

```diff
@@ -0,0 +1,159 @@
+import com.sap.piper.CloudPlatform
+import com.sap.piper.DeploymentType
+import com.sap.piper.k8s.ContainerMap
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.Utils
+import com.sap.piper.JenkinsUtils
+
+import groovy.transform.Field
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String STEP_NAME = getClass().getName()
+
+@Field Set GENERAL_CONFIG_KEYS = [
+     /** Defines the targets to deploy on cloudFoundry or neo.*/
+    'cfTargets',
+    'neoTargets'
+]
+
+@Field Set STEP_CONFIG_KEYS = []
+
+@Field Set PARAMETER_KEYS = GENERAL_CONFIG_KEYS.plus([
+     'stage',
+     /** Defines the deployment type */
+     'enableZeroDowntimeDeployment',
+    /** only for neo deployment*/
+    'source'
+])
+
+
+void call(parameters = [:]) {
+
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+
+        def index = 1
+        def deployments = [:]
+        def stageName = env.STAGE_NAME
+        def enableZeroDowntimeDeployment = parameters.enableZeroDowntimeDeployment
+
+        def script = checkScript(this, parameters) ?: this
+
+        def utils = parameters.utils ?: new Utils()
+        def jenkinsUtils = parameters.jenkinsUtils ?: new JenkinsUtils()
+ 
+        Map config = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)

```

Comment:
Is there any need for sanity checks (`withMandatoryProperty`) in order to fail fast in case there is some missing configuration?

Reply:
No